### DISCUSSION TITLE
script: more than 16384 chars possible?

### DISCUSSION THREAD

**Discussion by 2000ede** [arendst/Tasmota#20403] [Q&A] [answered]
I would like to have more than 16384 characters available in Tasmota Script (ESP32).
Is there already an option here today?
If not, can you make the request?

**Comment by gemu2015** [comment, marked as answer]
you can increase at cost of RAM as much as you like

#define UFSYS_SIZE  X

with larger code sizes you should better use external script editor because the GUI Window uses too much additional RAM.

you can also use code file subroutines or html file subroutines and such have arbitrary large programs

**Reply by jkwim** [reply]
>> you can also use code file subroutines or html file subroutines and such have arbitrary large programs

Could you elaborate more on how to use this method please?

**Comment by gemu2015** [comment]
its in the docs:
fe("/sub_script.txt")
execute a file named e.g. sub_script.txt.

needs #define USE_SCRIPT_FATFS_EXT

in all web sections
%/html_sub.txt
execute a file with html and javascript code.

**Reply by jkwim** [reply]
Thanks! Will try this out.
